torch/csrc/autograd/anomaly_mode.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+namespace torch { namespace autograd {
+
+struct AnomalyMode {
+  static bool is_enabled() {
+    return _enabled;
+  }
+  static void set_enabled(bool enabled) {
+    _enabled = enabled;
+  }
+
+private:
+  static bool _enabled;
```

Comment:
This should probably be thread local, or at least protected from being used in multiple threads.